www/js/common.js

```diff
@@ -1,100 +1,45 @@
-                                    leafletData, CommonGraph,Config) {
-  console.log("controller CommonCtrl called");
-
-  var db = window.cordova.plugins.BEMUserCache;
-  $scope.mapCtrl = {};
-
-  angular.extend($scope.mapCtrl, {
-    defaults : Config.getMapTiles(),
-    events: {
-      map: {
-        enable: ['zoomstart', 'drag', 'click', 'mousemove', 'contextmenu'],
-        logic: 'emit'
+.config(function($stateProvider, $ionicConfigProvider, $urlRouterProvider) {
+  $stateProvider
+
+  .state('root.main.common.map', {
+    url: '/map',
+    views: {
+      'menuContent': {
+        templateUrl: 'templates/common/map.html',
+        controller: 'CommonMapCtrl'
       }
     }
-  });
-
-  $scope.$on('leafletDirectiveMap.click', function(event){
-     alert('click '+JSON.stringify(event)+' detected');
-  });
-
-  $scope.$on('leafletDirectiveMap.contextmenu', function(event){
-     alert('conextmenu '+JSON.stringify(event)+' detected');
-  });
-
-  var onEachFeature = function(feature, layer) {
-    console.log("onEachFeature called with "+JSON.stringify(feature));
-    console.log("type is "+feature.geometry.type);
-    var popupContent = "<p>I started out as a GeoJSON " +
-                    feature.geometry.type + ", but now I'm a Leaflet vector!</p>";
-    layer.on('click', function(e) { console.log("layer clicked"); alert(feature.geometry.type) } );
-    // layer.bindPopup(popupContent);
-    /*
-    switch(feature.geometry.type) {
-      case "Point": layer.bindPopup(popupContent); break;
-      case "LineString": layer.bindPopup(popupContent); break;
-    }
-    */
-  }
-
-  $scope.refreshMap = function() {
-      CommonGraph.updateCurrent();
-  };
-
-  $scope.$on(CommonGraph.UPDATE_DONE, function(event, args) {
-    $scope.$apply(function() {
-        $scope.mapCtrl.geojson = {}
-        $scope.mapCtrl.geojson.data = CommonGraph.data.geojson;
-        $scope.mapCtrl.geojson.onEachFeature = onEachFeature;
-    });
-  });
-
-  $scope.refreshMap();
-
-  // TODO: Refactor into common code - maybe a service?
-  // Or maybe pre-populate from the server
-
-  /*
-  $scope.getDisplayName = function(place_feature) {
-    var responseListener = function(data) {
-      var address = data["address"];
-      var name = "";
-      if (address["road"]) {
-        name = address["road"];
-      } else if (address["neighbourhood"]) {
-        name = address["neighbourhood"];
+  })
+
+  .state('root.main.common.list', {
+    url: '/list',
+    views: {
+      'menuContent': {
+        templateUrl: 'templates/common/list.html',
+        controller: 'CommonListCtrl'
       }
-        if (address["city"]) {
-          name = name + ", " + address["city"];
-        } else if (address["town"]) {
-          name = name + ", " + address["town"];
-        } else if (address["county"]) {
-            name = name + ", " + address["county"];
+    }
+  })
+
+  .state('root.main.common.place-detail', {
+    url: '/place/:placeId',
```

Comment:
Are we going to have a list of places and when we press on an item from that list it will direct to a more detailed information about the place?